### PR TITLE
Fix sorting of persons and places to be case insensitive for names (EXPOSUREAPP-4488)(closes #2012)

### LINKED ISSUE #2012: Contacts and places are sorted case insensitive
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only (cannot check on Apple hardware)
* [X] Bug is not already reported in another issue

## Describe the bug

Contacts and places are sorted case insensitive. That makes it harder to find a contact again, if names are (accidentally) created with mixed lowercase and uppercase characters. UI elements sorted alphabetically should generally be case insensitive in all applications.

## Steps to reproduce the issue

1. Create contact "B".
2. Create contact "A1".
3. Create contact "a2".

## Expected behaviour

Contacts should be sorted A1, a2, B (case insensitive). Contacts are sorted A1, B, a2 (case sensitive).
The same issue exists for places.

## Possible Fix

If that is a Java collection being sorted, a second argument String.CASE_INSENSITIVE for the Collections.sort() call would be an appropriate fix.

---
Internal Tracking ID: [EXPOSUREAPP-4488](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4488)

### PR DESCRIPTION
**Pull request by PhilippNowak96**
### Description

As explained in #2012 the names of persons and places are currently sorted case sensitive in the contact diary resulting in lowercase names being added after **all** uppercase ones. This can lead to confusion especially for long name lists.

#### Current behavior

old-2012

#### New behavior (after fix)

new-2012

### Steps to reproduce

(stolen from #2012 😄 )

1. Create contact "B".
2. Create contact "A1".
3. Create contact "a2".
4. See "a2" being shown after "B"

### Further notes

I fixed this issue for persons and locations in the detail screen as well as for the contact diary overview. In addition I added some smaller unit tests for the sorting extension functions for names and places.

---
Internal Tracking ID: [EXPOSUREAPP-4488](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4488)

### CODE REVIEW

**Comment by dsarkar** [comment]
Dear @Bananeweizen, @PhilippNowak96, and community

Thank you for your contributions. We have created a ticket for the bug in the internal Jira system. Any further developments will be notified here in this Github issue.
  
Best wishes,
DS

---
Corona-Warn-App Open Source Team
Internal Tracking ID: [EXPOSUREAPP-4488](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4488)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryPersonEncounter.kt [hunk 1]
```diff
@@ -9,4 +10,4 @@ interface ContactDiaryPersonEncounter {
 }
 
 fun List<ContactDiaryPersonEncounter>.sortByNameAndIdASC(): List<ContactDiaryPersonEncounter> =
-    this.sortedWith(compareBy({ it.contactDiaryPerson.fullName }, { it.contactDiaryPerson.personId }))
+    this.sortedWith(compareBy({ it.contactDiaryPerson.fullName.toLowerCase(Locale.ROOT) }, { it.contactDiaryPerson.personId }))
```
Comment (thomasaugsten) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryPersonEncounter.kt:13]: Line is too long for Sonar Check

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryLocationVisit.kt [hunk 2]
```diff
@@ -9,4 +10,4 @@ interface ContactDiaryLocationVisit {
 }
 
 fun List<ContactDiaryLocationVisit>.sortByNameAndIdASC(): List<ContactDiaryLocationVisit> =
-    this.sortedWith(compareBy({ it.contactDiaryLocation.locationName }, { it.contactDiaryLocation.locationId }))
+    this.sortedWith(compareBy({ it.contactDiaryLocation.locationName.toLowerCase(Locale.ROOT) }, { it.contactDiaryLocation.locationId }))
```
Comment (thomasaugsten) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryLocationVisit.kt:13]: Line is too long for Sonar Check

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryPersonEncounter.kt (hunk 1 above)
Comment (PhilippNowak96) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryPersonEncounter.kt]: Fixed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryLocationVisit.kt (hunk 2 above)
Comment (PhilippNowak96) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/model/ContactDiaryLocationVisit.kt]: Fixed

**Review by d4rken — APPROVED** [review]
LGTM, thank you :heart:

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**